modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/artifact/ArtifactFactoryUtils.java

```diff
@@ -96,7 +96,7 @@ public static void withArtifactMuleContext(DeployableArtifact artifact, CheckedC
    * @since 4.2
    */
   public static Optional<MuleContext> getMuleContext(DeployableArtifact artifact) {
-    return artifact != null ? artifact.getRegistry().lookupByType(MuleContext.class) : empty();
+    return artifact != null && artifact.getRegistry() != null ? artifact.getRegistry().lookupByType(MuleContext.class) : empty();
```

Comment:
is this something that can really happen?

Reply:
still open

Reply:
The reasoning behind adding this additional check is the same as [here](https://github.com/mulesoft/mule/pull/9905#discussion_r559782412). Since `getRegistry()` can return null I think this method could eventually throw a `NullPointerException`